### PR TITLE
GUI Health

### PR DESCRIPTION
**Pull request by fredrix96**
Make sure that it looks fine in FredriksTestScene and Game and that you are able to gain health by the health gain upgrade in Game

### CODE REVIEW

**Review by jakkeez — DISMISSED** [review]

**Review by Whixx — CHANGES_REQUESTED** [review]
Crashes on death. Make it not.

**Review by Whixx — DISMISSED** [review]

**Review by Whixx — DISMISSED** [review]

**Review by Whixx — DISMISSED** [review]

**Review by jocke1995 — CHANGES_REQUESTED** [review]
Maxhealth doesn't update when buying hp in shopScene. Pls fix, I would also see the gui to be just a tiny bit smaller but I am not an artist so dont do that if you dont want to :))))))))))))))))))))))))

File: StortSpelProjekt/Game/src/Gamefiles/GameGUI.cpp [hunk 1]
```diff
@@ -6,42 +6,136 @@
 
 GameGUI::GameGUI()
 {
-	m_OldHealth = 0;
+	m_OldMaxHealth = 0;
+	m_OldCurrHealth = 0;
```
Comment (Whixx) [inline StortSpelProjekt/Game/src/Gamefiles/GameGUI.cpp]: This name does not make sense.

**Review by Whixx — CHANGES_REQUESTED** [review]

**Review by Whixx — DISMISSED** [review]

File: StortSpelProjekt/Game/src/Gamefiles/GameGUI.h [hunk 2]
```diff
@@ -14,7 +14,13 @@ class GameGUI
 
 private:
 	int m_OldHealth;
+	int m_OldMaxHealth;
 	int m_OldMoney;
+	float m_OldHealthLength;
+	Scene* m_OldScene;
```
Comment (jocke1995) [inline StortSpelProjekt/Game/src/Gamefiles/GameGUI.h]: m_pOldScene

**Review by jocke1995 — CHANGES_REQUESTED** [review]
Fix

**Review by jocke1995 — APPROVED** [review]
good

**Review by Whixx — APPROVED** [review]

**Pull request merged**